src/librustc/ty/layout.rs

```diff
@@ -1513,6 +1531,59 @@ impl<'a, 'gcx, 'tcx> Layout {
             }
         }
     }
+
+    pub fn field_offset(&self,
+                        dl: &TargetDataLayout,
+                        i: usize,
+                        variant_index: Option<usize>)
+                        -> Size {
+        match *self {
+            Scalar { .. } |
+            CEnum { .. } |
+            UntaggedUnion { .. } |
+            RawNullablePointer { .. } => {
```

Comment:
> debug_assert i==0, perhaps?

You mean here? I might just take this out, except for `UntaggedUnion`. It was meant for something I abandoned.